src/nvim/ex_getln.c

```diff
@@ -4308,11 +4306,12 @@ in_history (
       last_i = i;
     }
     history[type][i].hisnum = ++hisnum[type];
-    history[type][i].viminfo = FALSE;
     history[type][i].hisstr = str;
-    return TRUE;
+    history[type][i].timestamp = os_time();
+    history[type][i].additional_elements = NULL;
```

Comment:
We have to recreate the found entry in the last entry, but we only remember pointer `hisstr`.
Is `additional_elements` always `NULL` for a history entry? Otherwise we need to remember this value too:
```diff
diff --git a/src/nvim/ex_getln.c b/src/nvim/ex_getln.c
index 88b9a21..f8e5c9c 100644
--- a/src/nvim/ex_getln.c
+++ b/src/nvim/ex_getln.c
@@ -4299,6 +4299,7 @@ in_history (
 
   if (last_i >= 0) {
     str = history[type][i].hisstr;
+    list_T* list = history[type][i].additional_elements;
     while (i != hisidx[type]) {
       if (++i >= hislen)
         i = 0;
@@ -4308,7 +4309,7 @@ in_history (
     history[type][i].hisnum = ++hisnum[type];
     history[type][i].hisstr = str;
     history[type][i].timestamp = os_time();
-    history[type][i].additional_elements = NULL;
+    history[type][i].additional_elements = list;
     return true;
   }
   return false;
```

But this change makes a test fail:
```
Failure → test/functional/shada/compatibility_spec.lua @ 346
ShaDa forward compatibility support code works with history item with BOOL additional value in list
test/functional/shada/compatibility_spec.lua:368: Expected objects to be the same.
Passed in:
(number) 3
Expected:
(number) 2

stack traceback:
	test/functional/shada/compatibility_spec.lua:368: in function <test/functional/shada/compatibility_spec.lua:346>

```